### PR TITLE
rpc: adopt logging in circuitbreaker

### LINKED ISSUE #33657: rpc: log when breaker trips, including the error
We've seen indications (#31361, #33549) that circuit breakers sometimes open in situations in which it doesn't seem to make a whole lot of sense. If a breaker trips, we'd like to know the error that tripped it and the callsite.

### PR DESCRIPTION
**Pull request by ajwerner**
Adopts changes to the circuitbreaker package to enable logging. 
Fixes #33657

Release note: None

### CODE REVIEW

File: pkg/rpc/breaker.go [hunk 1]
```diff
@@ -89,10 +91,36 @@ func newBackOff(clock backoff.Clock) backoff.BackOff {
...
+func newBreaker(ctx context.Context, name string, clock clock.Clock) *circuit.Breaker {
 	return circuit.NewBreakerWithOptions(&circuit.Options{
 		BackOff:    newBackOff(clock),
 		Clock:      clock,
 		ShouldTrip: circuit.ThresholdTripFunc(1),
+		Logger:     breakerLogger{ctx},
 	})
 }
+
+// *clogLogger implements the raft.Logger interface. Note that all methods
```
Comment (tbg) [inline pkg/rpc/breaker.go]: This comment needs an update.

File: pkg/rpc/breaker.go [hunk 2]
```diff
@@ -89,10 +91,36 @@ func newBackOff(clock backoff.Clock) backoff.BackOff {
 	return b
 }
 
-func newBreaker(clock clock.Clock) *circuit.Breaker {
+func newBreaker(ctx context.Context, name string, clock clock.Clock) *circuit.Breaker {
```
Comment (tbg) [inline pkg/rpc/breaker.go]: `name` looks unused, I think you want to plumb it into the logger.

**Review by tbg — APPROVED** [review]
LGTM after comments. Thanks for getting to this so quickly.

File: pkg/rpc/breaker.go (hunk 1 above)
Comment (ajwerner) [reply pkg/rpc/breaker.go]: Done.

File: pkg/rpc/breaker.go (hunk 2 above)
Comment (ajwerner) [reply pkg/rpc/breaker.go]: Done.

**Review by tbg — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33676)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/rpc/breaker.go, line 105 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/33676#-LVylSzx3jz9zA3ZgKeF:-LVylSzy5y6E-DXlz4wL:b-xksgof) ([raw file](https://github.com/cockroachdb/cockroach/blob/40ded36db3577ff1b609434dd270a894b41ee12c/pkg/rpc/breaker.go#L105)):*
> ```Go
> 
> // breakerLogger implements circuit.Logger to expose logging from the
> // circuitbreaker package. Debugf is logged with a vmodule level of 1 so to see
> ```

Looks like we need V(3), not V(1).

What's the output that will be logged (from `Infof`) when the breaker trips? I assume it'll include the name.

**Review by ajwerner — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33676)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/rpc/breaker.go, line 105 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/33676#-LVylSzx3jz9zA3ZgKeF:-LVym82a9mHfW_6dST1z:b-hg4xch) ([raw file](https://github.com/cockroachdb/cockroach/blob/40ded36db3577ff1b609434dd270a894b41ee12c/pkg/rpc/breaker.go#L105)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

Looks like we need V(3), not V(1).

What's the output that will be logged (from `Infof`) when the breaker trips? I assume it'll include the name.
</blockquote></details>

I really need to be paying more attention :/, updating the comment now. The circuitbreaker logging always has `circuitbreaker: <name> ...`

**Review by ajwerner — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33676)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/rpc/breaker.go, line 105 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/33676#-LVylSzx3jz9zA3ZgKeF:-LVynFM672QQFNZnsiyi:b-6domsh) ([raw file](https://github.com/cockroachdb/cockroach/blob/40ded36db3577ff1b609434dd270a894b41ee12c/pkg/rpc/breaker.go#L105)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

I really need to be paying more attention :/, updating the comment now. The circuitbreaker logging always has `circuitbreaker: <name> ...` 
</blockquote></details>

Went with 2

**Pull request merged**